### PR TITLE
STN-172: Local Footer

### PR DESCRIPTION
**Pull request by kaseybon**
# READY FOR REVIEW

## Summary
This pull request adds the local footer component with a default light variant as well as a theme option for a dark variant. Footer blocks use the Decanter flexbox grid system for layout. This grid system includes responsive classes and are added when configuring blocks.

- This PR also includes a z-index fix where the lockup was overlaying the admin bar at small-medium screen sizes.

**Tested in the following browsers:**
- Safari
- Chrome, Windows & Mac
- Firefox, Windows & Mac
- Edge
- IE 11

**Default:**
Faculty___Economics

**Dark Variant:**
Faculty___Economics

In order to ensure that blocks are styled correctly the following items have been refactored:
- Move font styles from paragraph tag into the body element.
- Refactor button class to work with anchor tags.
- Style Font Awesome elements.

✅ Design reviewed and approved by @meranicosme 

## Need Review By (Date)
02/24

## Urgency
medium

## Steps to Test
1. Run `npm run build:sass` to compile assets
1. Run `npm test` to confirm there are no errors
1. Styles should match the [design mockups](https://www.figma.com/file/toNZnFLadtpNpG5dPzNGWz/%E2%9A%99%EF%B8%8F-Colorful-Theme---Working?node-id=3324%3A2505) for both light and dark variants.
1. Configure blocks to use [Flexbox Grid Classes](https://decanter.stanford.edu/page/layouts-grid-system/) for layout styles.
1. Confirm that the following block types match the screenshots above:
    - Menu block
    - Custom block with text
    - Custom block with image
    - Custom block with buttons
    - Custom block with Font Awesome icons.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

### CODE REVIEW

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr496-or9mvzcwkox6cxuuromxhnf1avgqdy8x.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e542905a22dc6804b798bdb

File: docroot/themes/humsci/humsci_basic/theme-settings.php [hunk 1]
```diff
@@ -121,6 +121,22 @@ function humsci_basic_form_system_theme_settings_alter(array &$form, FormStateIn
...
+  $form['options_settings']['humsci_basic_local_footer']['local_footer_variant_classname'] = [
+    '#type' => 'select',
+    '#title' => t('Local Footer Variant'),
+    '#options' => [
+      'default' => '- Default -',
+      'dark' => t('Dark'),
+    ],
+    '#default_value' => theme_get_setting('local_footer_variant_classname'),
+  ];
+
```
Comment (bryanbraun) [inline docroot/themes/humsci/humsci_basic/theme-settings.php]: Currently this theme setting is in `humsci_basic` so it's providing the theme setting to all the sub-themes (including traditional and airy). That's been fine for the decanter components (brand bar / global footer), but I'm guessing it's not what we want for this (though we should probably check with Alex).

If this is supposed to be colorful specific, we need to move some of these things into the colorful sub theme.

**Review by heatherdesigns — CHANGES_REQUESTED** [review]
This is looking fantastic! Great work!

I noticed Bryan left a comment regarding the location of the local footer variant setting.

When tabbing through the dark variant, the focus state has a box-shadow on the link that makes it hard to read. I'd change the focus state of the link to match the hover state so that it is readable.
focus hard to read content

File: docroot/themes/humsci/humsci_basic/theme-settings.php (hunk 1 above)
Comment (kaseybon) [reply docroot/themes/humsci/humsci_basic/theme-settings.php]: I think this makes sense. I moved the variant to the colorful theme settings and tested the change. I installed the Airy theme and confirmed that:

- The variant only shows up for the Colorful theme
- Other themes do not break when switching themes. The airy theme gets the default class only.

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr496-or9mvzcwkox6cxuuromxhnf1avgqdy8x.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e542905a22dc6804b798bdb

**Review by heatherdesigns — APPROVED** [review]
Excellent! :tada:

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr496-or9mvzcwkox6cxuuromxhnf1avgqdy8x.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e542905a22dc6804b798bdb

**Pull request merged**